### PR TITLE
Add route_sort_order field to route in gtfs spec

### PR DESCRIPTION
**Pull request by evansiroky**
The field `route_sort_order` was missing from the route table in the gtfs yml spec.  This fixes that.

This depends on https://github.com/conveyal/gtfs-lib/pull/154 being used by datatools-server to fully work.

### CODE REVIEW

**Review by landonreed — COMMENTED** [review]
Any reason why the values are double quoted while the other yml string values aren't?

**Comment by evansiroky** [comment]
Was just copy pasting from some other area that had those quotes.  I removed them.  Shall I remove all the others as well while I'm at it?

**Comment by landonreed** [comment]
Oh, gotcha. Actually, it's fine to leave them in. I think there was some unquoted help content at one point that was causing issues, so we might as well just leave it in. Thanks.

**Review by landonreed — CHANGES_REQUESTED** [review]
@evansiroky, could you replace the `NUMBER` type here with the `POSITIVE_INT` field type? We want to prevent users from inputting a double, which causes the server to spit back the update. While you're at it, I think we're missing a validation check for that type, so it would be good if you could add that to this switch statement:

https://github.com/catalogueglobal/datatools-ui/blob/3b230bba85d556415f76e76f09f2a5a1c58cb1b5/lib/editor/util/validation.js#L240-L246

**Comment by landonreed** [comment]
@evansiroky, could you also move the field into a different position in the yml list? Adding it to the bottom has it awkwardly appear below the image upload. I think a more appropriate location would be next to `route_type` (that field's width would need to change to 6).
image

**Comment by evansiroky** [comment]
Order changed and validation for POSITIVE_INT and POSITIVE_NUM added.

File: lib/editor/util/validation.js [hunk 1]
```diff
@@ -246,6 +252,35 @@ export function validate (
     case 'COLOR':
     case 'POSITIVE_INT':
     case 'POSITIVE_NUM':
+      if (isRequiredButEmpty) {
```
Comment (landonreed) [inline lib/editor/util/validation.js]: Wouldn't this new check apply to all of the cases above? I think you only meant for it to apply to POSITIVE_INT and POSITIVE_NUM, right?

Comment (evansiroky) [reply lib/editor/util/validation.js]: oops

Comment (evansiroky) [reply lib/editor/util/validation.js]: Fixed that

File: gtfs.yml [hunk 2]
```diff
@@ -276,6 +276,11 @@
       text: Funicular
     columnWidth: 12
     helpContent: The route_type field describes the type of transportation used on a route. Valid values for this field are...
+  - name: route_sort_order
```
Comment (landonreed) [inline gtfs.yml:279]: In my [other comment](https://github.com/catalogueglobal/datatools-ui/pull/264#issuecomment-442922044) I had mentioned having this field be displayed next to route_type and changing route_type's width to 6. I was also implying that route_sort_order should have a width of 6. I now realize my language wasn't super clear. Could you make that fix as well? Or if you already tried it and it looked weird, no worries.

File: lib/editor/util/validation.js [hunk 3]
```diff
@@ -233,12 +246,41 @@ export function validate (
         }
       }
       return false
+    case 'POSITIVE_INT':
+    case 'POSITIVE_NUM':
+      if (isRequiredButEmpty) {
```
Comment (landonreed) [inline lib/editor/util/validation.js]: I'm not crazy about the validation logic for these two types being so closely linked. What do you think about moving the common "number validation" logic into a callable method that gets applied to the separate cases? For example, something like:

```
case 'POSITIVE_INT':
  const isInvalid = checkPositiveNumber(args)
  if (isInvalid) return isInvalid // if not a positive number, return validation issue immediately
  // Continue to check for integer validity.
  // ....
case 'POSITIVE_NUM':
  return checkPositiveNumber()
```

**Review by landonreed — COMMENTED** [review]
Just a couple more items I have some minor feedback on.

File: gtfs.yml (hunk 2 above)
Comment (evansiroky) [reply gtfs.yml:279]: Ah, I see what you mean now.  They can fit side-by-side.  The updated `gtfs.yml` should have the desired changes.

File: lib/editor/util/validation.js (hunk 3 above)
Comment (evansiroky) [reply lib/editor/util/validation.js]: Refactored as suggested.

**Comment by evansiroky** [comment]
Refactored as suggested.

**Review by landonreed — APPROVED** [review]

**Pull request merged**